rhino/src/main/java/org/mozilla/javascript/typedarrays/ByteIo.java

```diff
@@ -162,6 +162,116 @@ public static void writeUint64(byte[] buf, int offset, long val, boolean littleE
         }
     }
 
+    public static Float readFloat16(byte[] buf, int offset, boolean littleEndian) {
+        int bits = doReadInt16(buf, offset, littleEndian) & 0xffff;
+
+        // Extract sign, exponent, and mantissa
+        int sign = (bits >>> 15) & 0x1;
+        int exponent = (bits >>> 10) & 0x1f;
+        int mantissa = bits & 0x3ff;
+
+        // Handle special cases
+        if (exponent == 0) {
+            if (mantissa == 0) {
+                // Zero
+                return Float.valueOf(sign == 0 ? 0.0f : -0.0f);
+            } else {
+                // Denormalized number
+                float value = (float) (mantissa / 1024.0 * Math.pow(2, -14));
+                return Float.valueOf(sign == 0 ? value : -value);
+            }
+        } else if (exponent == 31) {
+            if (mantissa == 0) {
+                // Infinity
+                return Float.valueOf(sign == 0 ? Float.POSITIVE_INFINITY : Float.NEGATIVE_INFINITY);
+            } else {
+                // NaN
+                return Float.valueOf(Float.NaN);
+            }
+        } else {
+            // Normalized number
+            float value = (float) ((1.0 + mantissa / 1024.0) * Math.pow(2, exponent - 15));
```

Comment:
this class does a lot of bit shifting, maybe we can should do this for / 1024.0 also?